crates/hyperqueue/src/common/cli.rs

```diff
@@ -1,31 +1,56 @@
+use crate::client::status::Status;
 use crate::common::arraydef::IntArray;
-use crate::transfer::messages::Selector;
+use crate::transfer::messages::{IdSelector, StatusSelector};
 use std::str::FromStr;
 
-pub enum SelectorArg {
+pub enum IdSelectorArg {
     All,
     Last,
     Id(IntArray),
 }
 
-impl FromStr for SelectorArg {
+impl FromStr for IdSelectorArg {
     type Err = anyhow::Error;
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
         match s {
-            "last" => Ok(SelectorArg::Last),
-            "all" => Ok(SelectorArg::All),
-            _ => Ok(SelectorArg::Id(IntArray::from_str(s)?)),
+            "last" => Ok(IdSelectorArg::Last),
+            "all" => Ok(IdSelectorArg::All),
+            _ => Ok(IdSelectorArg::Id(IntArray::from_str(s)?)),
         }
     }
 }
 
-impl From<SelectorArg> for Selector {
-    fn from(selector_arg: SelectorArg) -> Self {
-        match selector_arg {
-            SelectorArg::Id(array) => Selector::Specific(array),
-            SelectorArg::Last => Selector::LastN(1),
-            SelectorArg::All => Selector::All,
+impl From<IdSelectorArg> for IdSelector {
+    fn from(id_selector_arg: IdSelectorArg) -> Self {
+        match id_selector_arg {
+            IdSelectorArg::Id(array) => IdSelector::Specific(array),
+            IdSelectorArg::Last => IdSelector::LastN(1),
+            IdSelectorArg::All => IdSelector::All,
+        }
+    }
+}
+
+impl From<Option<IdSelectorArg>> for IdSelector {
```

Comment:
It was quite weird for me to see you create an `Option` and then call `into` it. It is not immediately obvious what these conversions do, so I would rather create explicit functions for them, instead of implementing `From`.